components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/tcp/core/TCPMultiIOHandler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * /
+ */
+
+package org.wso2.carbon.inbound.endpoint.protocol.tcp.core;
+
+import org.apache.http.nio.reactor.IOSession;
+import org.apache.log4j.Logger;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * per connection/session IO event handler. IO reactor triggers each
+ * event accordingly
+ */
+public class TCPMultiIOHandler extends TCPSourceHandler {
```

Comment:
Why this is extend by TCPSourceHandler? I think TCPMultiIOHandler need to extend by AbstractIODispatch or implement by IOEventDispatc and internally it need to call TCPSourceHandler.  Any special reason?